packages/server-wallet/src/chain-service/chain-service.ts

```diff
@@ -412,7 +412,25 @@ export class ChainService implements ChainServiceInterface {
     contract: Contract,
     channelId: string
   ): Promise<HoldingUpdatedArg> {
+    const blockNumber = await this.provider.getBlockNumber();
     const holding = BN.from(await contract.holdings(channelId));
+    const depositEvents = await contract.queryFilter(
+      {
+        address: contract.address,
```

Comment:
I am surprised that this is needed. Isn't `queryFilter` invoked on a contract object that is aware of its own address?